app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.java

```diff
@@ -0,0 +1,154 @@
+        return findFor(
+                0,
+                position,
+                schedule.pages(),
+                schedulePage -> (long) -schedulePage.dayId().hashCode(),
+                (schedulePage, positionInPage) -> schedulePage.events().get(positionInPage).numericId()
+        );
+    }
+
+    void updateWith(Schedule schedule, ScheduleViewPagerAdapter.OnEventClickedListener listener) {
+        this.schedule = schedule;
+        this.listener = listener;
+        notifyDataSetChanged();
+    }
+
+    @Override
+    @ItemViewType
+    public int getItemViewType(int position) {
+        return findFor(
+                0,
+                position,
+                schedule.pages(),
+                schedulePage -> ItemViewType.TYPE_HEADER,
+                (schedulePage, positionInPage) -> ItemViewType.TYPE_TALK
+        );
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, @ItemViewType int viewType) {
+        if (viewType == ItemViewType.TYPE_TALK) {
+            EventItemView itemView = (EventItemView) layoutInflater.inflate(R.layout.item_schedule_event_talk, parent, false);
+            return new EventViewHolder(itemView);
+        } else if (viewType == ItemViewType.TYPE_HEADER) {
+            return new HeaderViewHolder(layoutInflater.inflate(R.layout.item_search_header, parent, false));
+        } else {
+            throw new IllegalArgumentException("View type not supported: " + viewType);
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        if (holder instanceof EventViewHolder) {
+            Event event = findFor(
+                    0,
+                    position,
+                    schedule.pages(),
+                    schedulePage -> {
+                        throw new IndexOutOfBoundsException();
+                    },
+                    (schedulePage, positionInPage) -> schedulePage.events().get(positionInPage)
+            );
+
+            ((EventViewHolder) holder).updateWith(event, listener);
+        } else if (holder instanceof HeaderViewHolder) {
+            LocalDateTime date = findFor(
+                    0,
+                    position,
+                    schedule.pages(),
+                    SchedulePage::date,
+                    (schedulePage, positionInPage) -> {
+                        throw new IndexOutOfBoundsException();
+                    }
+            );
+            ((HeaderViewHolder) holder).updateWith(formatHeader(date));
+        }
+    }
+
+    private CharSequence formatHeader(LocalDateTime date) {
+        return date.toString("EEEE d");
+    }
+
+    @Override
+    public int getItemCount() {
+        return Lists.reduce(0, schedule.pages(), (count, page) -> count + page.events().size() + 1);
+    }
+
+    private <T> T findFor(int pagePosition, int position, List<SchedulePage> pages, Header<T> header, Row<T> row) {
```

Comment:
I would extract this on a separate repository class that returns us the list we need, so that we keep the adapter as stupid as we can